pyproject.toml

```diff
@@ -33,6 +33,11 @@ copyright-check = "True"
 copyright-author = "Canonical Ltd."
 copyright-regexp = "Copyright\\s\\d{4}([-,]\\d{4})*\\s+%(author)s"
 
+[tool.autoflake]
+check = true
+remove-all-unused-imports = true
```

Comment:
Great addition!

Reply:
pre-commit is a common practice is lots of repos. Nobody is pushed to use pre-commit if they do not want to, but if you want, then you just go for `pre-commit install`.

every linting tool provides a pre-commit hook, I think it is also a positive sign

the only change is that we provide an extra dependency, but I do not know how much of that is a drawback.

pre-commit provides a native support as hooks, you do not need to configure anything extra. It will run only on files added to staging. Do not see a point to reinvent the wheel when there is already a great community solution.

Reply:
I would prefer to just have a pre-commit git hook that you can copy in if you want. I see no reason to install extra software to do this.

Reply:
what is the sense to spend time creating and debugging hooks if we have out of the box solution?